File: src/main/Java/boj_step15/boj4948.java
import java.io.*;
import java.util.Arrays;

public class boj4948 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        boolean[] isPrime = new boolean[250_001];
        Arrays.fill(isPrime, true);
        for (int i = 2; i <= isPrime.length / 2; i++) {
            if (!isPrime[i]) continue;
            for (int j = i * 2; j < isPrime.length; j += i) isPrime[j] = false;
        }
        String line;
        while (!(line = reader.readLine()).equals("0")) {
            int n = Integer.parseInt(line);
            int ans = 0;
            for (int i = n + 1; i <= n * 2; i++) if (isPrime[i]) ans++;
            writer.append(ans + "\n");
        }
        writer.close();
    }
}

File: src/main/boj_step15/boj4948.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
        val r = BufferedReader(InputStreamReader(System.`in`))
        val w = BufferedWriter(OutputStreamWriter(System.out))
        val maxN = 250000
        var primes = Array(maxN + 1) { true }
        for (i in 2..maxN / 2) {
                if (!primes[i]) continue
                for (j in i * 2..maxN step i) primes[j] = false
        }
        while (true) {
                val k = r.readLine().toInt()
                if (k == 0) break
                var ans = 0
                for (i in k + 1..k * 2) if (primes[i]) ans++
                w.write(ans.toString())
                w.newLine()
        }
        w.close()
}
